### PR TITLE
Separate IndexedStorer and Storer Traits

### PR DESCRIPTION
**Pull request by howhelan**

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **howhelan**
You can assign the PR to them by writing `/assign @howhelan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/google-cloud-storage/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | 261542610709d3cb6ab7f29b73fb12fb422b5b95 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/abs-redact-crypto-pr-7-unit-test-qjwc4) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/storage.rs
Comment (ajpauwels) [inline src/storage.rs:33]: Should this a non-indexed mock?

**Review by ajpauwels — CHANGES_REQUESTED** [review]

**Comment by capt-haddock** [comment]
@howhelan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | f853ef52e80363ba1bc8172be5345b01bf95f598 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/abs-redact-crypto-pr-7-unit-test-rgsgt) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/storage.rs
Comment (howhelan) [reply src/storage.rs]: fixed by adding TypeStorer enum above IndexedTypeStorer and NonIndexedTypeStorer

**Review by ajpauwels — APPROVED** [review]

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ajpauwels](https://github.com/pauwels-labs/redact-crypto/pull/7#pullrequestreview-750650683 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/google-cloud-storage/OWNERS)~~ [ajpauwels]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
